fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryStreamingAggregationPlan.java

```diff
@@ -351,27 +367,60 @@ public Value getCompleteResultValue() {
 
     @Nonnull
     @Override
-    public PRecordQueryStreamingAggregationPlan toProto(@Nonnull final PlanSerializationContext serializationContext) {
-        final var builder = PRecordQueryStreamingAggregationPlan.newBuilder()
-                .setInner(inner.toProto(serializationContext))
-                .setAggregateValue(aggregateValue.toValueProto(serializationContext));
-        if (groupingKeyValue != null) {
-            builder.setGroupingKeyValue(groupingKeyValue.toValueProto(serializationContext));
+    public Message toProto(@Nonnull final PlanSerializationContext serializationContext) {
+        if (serializationMode == SerializationMode.TO_OLD) {
+            final var builder = PRecordQueryStreamingAggregationPlan.newBuilder()
+                    .setInner(inner.toProto(serializationContext))
+                    .setAggregateValue(aggregateValue.toValueProto(serializationContext));
+            if (groupingKeyValue != null) {
+                builder.setGroupingKeyValue(groupingKeyValue.toValueProto(serializationContext));
+            }
+            builder.setGroupingKeyAlias(groupingKeyAlias.getId())
+                    .setAggregateAlias(aggregateAlias.getId())
+                    .setCompleteResultValue(completeResultValue.toValueProto(serializationContext))
+                    .setIsCreateDefaultOnEmpty(false);
+            return builder.build();
+        } else {
+            final var builder = PRecordQueryStreamingAggregationPlan2.newBuilder()
+                    .setInner(inner.toProto(serializationContext))
+                    .setAggregateValue(aggregateValue.toValueProto(serializationContext));
+            if (groupingKeyValue != null) {
+                builder.setGroupingKeyValue(groupingKeyValue.toValueProto(serializationContext));
+            }
+            builder.setGroupingKeyAlias(groupingKeyAlias.getId())
+                    .setAggregateAlias(aggregateAlias.getId())
+                    .setCompleteResultValue(completeResultValue.toValueProto(serializationContext));
+            return builder.build();
         }
-        builder.setGroupingKeyAlias(groupingKeyAlias.getId())
-                .setAggregateAlias(aggregateAlias.getId())
-                .setCompleteResultValue(completeResultValue.toValueProto(serializationContext))
-                .setIsCreateDefaultOnEmpty(false);
-        return builder.build();
     }
 
     @Nonnull
     @Override
     public PRecordQueryPlan toRecordQueryPlanProto(@Nonnull final PlanSerializationContext serializationContext) {
-        return PRecordQueryPlan.newBuilder().setStreamingAggregationPlan(toProto(serializationContext)).build();
+        if (serializationMode == SerializationMode.TO_OLD) {
+            return PRecordQueryPlan.newBuilder().setStreamingAggregationPlan((PRecordQueryStreamingAggregationPlan)toProto(serializationContext)).build();
```

Comment:
It might make sense to just have to methods, `toProtoOld` and `toProtoNew`, which produce the correct types, and then you don't need to have to cast and you don't need to do another `if` statement on the `serializationMode`

Reply:
modified